src/backend/dx11/src/data.rs

```diff
@@ -206,54 +246,67 @@ pub fn map_bind(bind: Bind) -> D3D11_BIND_FLAG {
 
 pub fn map_access(access: memory::Access) -> D3D11_CPU_ACCESS_FLAG {
     let mut r = D3D11_CPU_ACCESS_FLAG(0);
-    if access.contains(memory::READ) { r = r | D3D11_CPU_ACCESS_READ }
-    if access.contains(memory::WRITE) { r = r | D3D11_CPU_ACCESS_WRITE }
+    if access.contains(memory::READ) {
+        r = r | D3D11_CPU_ACCESS_READ
+    }
+    if access.contains(memory::WRITE) {
+        r = r | D3D11_CPU_ACCESS_WRITE
+    }
     r
 }
 
 pub fn map_usage(usage: Usage) -> (D3D11_USAGE, D3D11_CPU_ACCESS_FLAG) {
     match usage {
-        Usage::GpuOnly => (D3D11_USAGE_DEFAULT,   D3D11_CPU_ACCESS_FLAG(0)),
+        Usage::GpuOnly => (D3D11_USAGE_DEFAULT, D3D11_CPU_ACCESS_FLAG(0)),
         Usage::Immutable => (D3D11_USAGE_IMMUTABLE, D3D11_CPU_ACCESS_FLAG(0)),
-        Usage::Dynamic => (D3D11_USAGE_DYNAMIC,   D3D11_CPU_ACCESS_WRITE),
-        Usage::Mappable(_access) => unimplemented!(),
+        Usage::Dynamic => (D3D11_USAGE_DYNAMIC, D3D11_CPU_ACCESS_WRITE),
+        Usage::Mappable(access) => {
+            if access.contains(memory::READ) {
+                panic!("DX does not support mapped memory for read access!");
```

Comment:
I guess you could check the requested access before calling this function (during buffer creation) or maybe return an option and map it to an error later ?